File: app/src/main/java/code/name/monkey/retromusic/mvp/contract/SearchContract.java
package code.name.monkey.retromusic.mvp.contract;

import code.name.monkey.retromusic.mvp.BasePresenter;
import code.name.monkey.retromusic.mvp.BaseView;

import java.util.ArrayList;


/**
 * Created by hemanths on 20/08/17.
 */

public interface SearchContract {
    interface SearchView extends BaseView<ArrayList<Object>> {

    }

    interface SearchPresenter extends BasePresenter<SearchView> {
        void search(String query);
    }
}

File: app/src/main/java/code/name/monkey/retromusic/mvp/contract/SearchContract.kt
package code.name.monkey.retromusic.mvp.contract

import code.name.monkey.retromusic.mvp.BasePresenter
import code.name.monkey.retromusic.mvp.BaseView
import java.util.*


/**
 * Created by hemanths on 20/08/17.
 */

interface SearchContract {
    interface SearchView : BaseView<ArrayList<Any>>

    interface SearchPresenter : BasePresenter<SearchView> {
        fun search(query: String?)
    }
}
